src/main/java/frc/robot/commands/AutoBalance.java

```diff
@@ -0,0 +1,117 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+
+public class AutoBalance extends CommandBase {
+    private double m_pitch;
+    private double m_error = 0.0d;
+    private double m_lastError = 0.0d;
+    private double m_change = 0.0d;
+    private double m_errorIntegral = 0.0d;
+    private double m_velOutput = 0.0d;
+
+    private double m_velMultipliedOutput = 0.0d;
+
+    // start (gives throttle) (may make it overshoot if too high)
+    private double m_P = 0.0031d;
+    // finicky (depends on situation) (within 5 to 3 degress of error)
+    private double m_I = 0.00d;
+    // good rule of thumb for d: m_d = m_p * 10
+    private double m_D = m_P * 10;
+
+    // speed control
+    private double m_clamp = 0.5;
+    private int m_balancedTimes = 0;
+    private double m_maxBalance = 20;
+    private double vel = 2.5;
+
+    private Drivetrain m_driveTrain;
+
+    public AutoBalance(Drivetrain driveTrain) {
+
+        m_driveTrain = driveTrain;
+        addRequirements(driveTrain);
+
+        SmartDashboard.putNumber("m_P", m_P);
+        SmartDashboard.putNumber("m_I", m_I);
+        SmartDashboard.putNumber("m_D", m_D);
+        SmartDashboard.putNumber("m_Clamp", m_clamp);
+
+        SmartDashboard.putNumber("vel", vel);
+        SmartDashboard.putNumber("maxbalance", m_maxBalance);
+    }
+
+    public int CheckNearestNumber(int back, int num, int front) {
+        var frontDiff = Math.abs(front - num);
+        var backDiff = Math.abs(back - num);
+        if (frontDiff < backDiff) {
+            return front;
```

Comment:
Do not do more than one return path please

Reply:
updated